### PR TITLE
[kjobctl] Add list slurm command.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Add list slurm command on kjobctl.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by mwielgus** [comment]
Is it just copy+search+replace version of the other list commands or are there fragments significantly different from others?

**Comment by mbobrovskyi** [comment]
We've added the `kjobctl.x-k8s.io/mode` label, which allows us to filter and display only Slurm jobs when viewing the Slurm list and only Batch jobs when viewing the Job list.

File: cmd/experimental/kjobctl/pkg/cmd/list/list_slurm.go
Comment (alculquicondor) [inline cmd/experimental/kjobctl/pkg/cmd/list/list_slurm.go:1]: isn't this a lot of repeated code from list_job?

Good for now, but maybe we can refactor some shared bits in a follow up?

**Pull request merged**